File: 10TypeScript/Ex1_3.ts
class Ex1_3 {
    calcLcd(dividend: number, divisor: number): number {
        let remainder: number = dividend % divisor;
        if (remainder === 0) return divisor;
        return this.calcLcd(divisor, remainder);
    }

    main() {
        var dividend = parseInt(process.argv[2]);
        var divisor = parseInt(process.argv[3]);
        
        return this.calcLcd(dividend, divisor);
    }
}

let app = new Ex1_3();

console.log(app.main());

File: 09Scala/Ex1_3.scala
object Ex1_3 {
  def calcLcd(dividend: Int, divisor: Int): Int = {
    var remainder = dividend % divisor
    return if (remainder == 0) divisor else calcLcd(divisor, remainder)
  }

  def main(args: Array[String]): Unit = {
    var result = calcLcd(args(0).toInt, args(1).toInt)
    println(s"$result")
  }
}

